File: _python/350.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        record = dict.fromkeys(nums1,0)
        for x in nums1:
            record[x]+=1
        res = []
        for x in nums2:
            if record.get(x,0)>0:
                res.append(x)
                record[x]-=1
        return res


if __name__ == '__main__':
    n1=[1,2,2,1]
    n2=[2,2]
    print Solution().intersect(n1,n2)

File: _cpp/answer/350.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace std;

namespace p350 {
    class Solution {
    public:
        vector<int> intersect(vector<int> &nums1, vector<int> &nums2) {
            map<int, int> record;
            for (int i:nums1)
                record[i]++;
            vector<int> res;
            for (int j:nums2)
                if (record[j] > 0) {
                    res.push_back(j);
                    record[j]--;
                }

            return res;
        }
    };

}